Test failure looks real or at least newly flaky

```
======================================================================
  FAIL: corehq.apps.api.tests.core_api:TestApiThrottle.test_throttle_allowlist
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/usr/local/lib/python3.9/unittest/mock.py", line 1336, in patched
      return func(*newargs, **newkeywargs)
    File "/mnt/commcare-hq-ro/corehq/apps/api/tests/core_api.py", line 721, in test_throttle_allowlist
      self.assertEqual(response.status_code, 429)
  AssertionError: 200 != 429
```

--- a/corehq/apps/api/tests/core_api.py
+++ b/corehq/apps/api/tests/core_api.py
@@ -714,8 +714,8 @@ class TestApiThrottle(APIResourceTest):
     def test_throttle_allowlist(self):
         """Test that the allowlist toggle allows all traffic through
         """
-        with patch('corehq.apps.api.resources.meta.BaseThrottle.should_be_throttled') as should_be_throttled:
-            should_be_throttled.return_value = True
+        with patch('corehq.apps.api.resources.meta.api_rate_limiter.allow_usage') as allow_access:
+            allow_access.return_value = False
 
             response = self.client.get(self.endpoint)
             self.assertEqual(response.status_code, 429)

